storage/src/main/java/tech/pegasys/artemis/storage/FixedStateChainStorageServer.java

```diff
@@ -0,0 +1,130 @@
+
+public class FixedStateChainStorageServer implements ChainStorage {
+  private static final ALogger STDOUT = new ALogger("stdout");
+
+  private final EventBus eventBus;
+  private final ArtemisConfiguration config;
+
+  public FixedStateChainStorageServer(EventBus eventBus, ArtemisConfiguration config) {
+    this.eventBus = eventBus;
+    this.config = config;
+    eventBus.register(this);
+  }
+
+  @Subscribe
+  public void onNodeStart(NodeStartEvent nodeStartEvent) {
+    final String interopStartState = config.getInteropStartState();
+    if (config.getInteropActive() && interopStartState != null) {
+      try {
+        STDOUT.log(Level.INFO, "Loading initial state from " + interopStartState, Color.GREEN);
+        final DBStoreValidEvent dbStoreValidEvent =
+            loadInitialState(Bytes.wrap(Files.readAllBytes(new File(interopStartState).toPath())));
+        this.eventBus.post(dbStoreValidEvent);
+      } catch (final IOException e) {
+        throw new IllegalStateException("Failed to load initial state", e);
+      }
+    }
+  }
+
+  DBStoreValidEvent loadInitialState(final Bytes beaconStateData) {
+    final BeaconStateWithCache initialBeaconState = loadBeaconState(beaconStateData);
+
+    final Store initialStore =
+        createInitialStore(
+            initialBeaconState, UnsignedLong.valueOf(config.getInteropGenesisTime()));
+    return new DBStoreValidEvent(initialStore, initialBeaconState.getSlot());
+  }
+
+  private static Store createInitialStore(
+      final BeaconStateWithCache initialState, final UnsignedLong genesisTime) {
+    return get_genesis_store(initialState);
+  }
+
+  public static Store get_genesis_store(BeaconStateWithCache genesis_state) {
+    BeaconBlock genesis_block = new BeaconBlock(genesis_state.hash_tree_root());
+    Bytes32 root = genesis_block.signing_root("signature");
+    Checkpoint justified_checkpoint = new Checkpoint(UnsignedLong.valueOf(GENESIS_EPOCH), root);
+    Checkpoint finalized_checkpoint = new Checkpoint(UnsignedLong.valueOf(GENESIS_EPOCH), root);
+    ConcurrentHashMap<Bytes32, BeaconBlock> blocks = new ConcurrentHashMap<>();
+    ConcurrentHashMap<Bytes32, BeaconState> block_states = new ConcurrentHashMap<>();
+    ConcurrentHashMap<Checkpoint, BeaconState> checkpoint_states = new ConcurrentHashMap<>();
+    blocks.put(root, genesis_block);
+    block_states.put(root, new BeaconStateWithCache(genesis_state));
+    checkpoint_states.put(justified_checkpoint, new BeaconStateWithCache(genesis_state));
+    return new Store(
+        genesis_state.getGenesis_time(),
+        justified_checkpoint,
+        finalized_checkpoint,
+        blocks,
+        block_states,
+        checkpoint_states);
+  }
+
+  private static BeaconStateWithCache loadBeaconState(final Bytes data) {
+    final BeaconState initialState;
+    initialState = SimpleOffsetSerializer.deserialize(data, BeaconState.class);
+    return new BeaconStateWithCache(
```

Comment:
Why not use the deepCopy method here?

Reply:
Because we're converting from a `BeaconState` to a `BeaconStateWithCache` - `deepCopy` requires a `BeaconStateWithCache` as input. I did just move it to a static method in `BeaconStateWithCache` though.

Reply:
Oh makes sense.